src/vswitch.c

```diff
@@ -37,8 +40,12 @@ SEXP vswitchR(SEXP x, SEXP values, SEXP outputs, SEXP na, SEXP nthreads) {
   SEXPTYPE type_x = UTYPEOF(x);
   SEXPTYPE type_na = UTYPEOF(na);
   SEXPTYPE type_values = UTYPEOF(values);
+  const bool pchkenc = asLogical(chkenc);
   const bool nonna = !isNull(na);
   const bool nalen = (len_na==1);
+  if (!len_values || !len_x) {
+    error("Argument 'x' and 'values' connot be zero-length vector.");
```

Comment:
I think 0-length x should return a 0-length value. This is also the current behavior of vswitch().

```r
> kit::vswitch(character(), "a", 1)
numeric(0)
```

Reply:
above behaviour should remain now

Reply:
Great thanks!